R/observations.R

```diff
@@ -52,11 +55,23 @@ prepare_data <- function(x, ...) {
 #' @rdname prepare_data
 #' @name prepare_data
 #' @export
-prepare_data.empirical <- function(x, seed = 1, n_samples = 1e3, ...) {
+prepare_data.empirical <- function(x, seed = 1, n_samples = 1e3, index_features = NULL, ...) {
+
+  # Get distance matrix ----------------
+  if (is.null(index_features)) {
+    index_features <- x$X[, .I]
+  }
+
+  x$D <- distance_matrix(
+    x$x_train,
+    x$x_test,
+    x$X$features[index_features]
+  )
 
   # Setup
+  if (!is.null(seed)) set.seed(seed)
   n_col <- nrow(x$x_test)
-  no_empirical <- nrow(x$S)
+  no_empirical <- nrow(x$S[index_features, ])
```

Comment:
I think you need drop=F here to handle cases where length(index_features)=1. Shouldn't a test have captured such a problem?

Reply:
`index_features` should never be an element less than 2. I'll need to make an small change to make sure that this is the case.

Reply:
This should fix it. c7fe376